### ISSUE TITLE
Working with Gitlab groups and subgroups

### ISSUE THREAD

**Issue by dark452** [tiged/tiged#23]
Hi!,

I'm trying to use tiged with a on-premises _Gitlab_ implementation and i'm not able to made it work with **groups** that have **subgroups** inside.

For example i have the following structure:

ParentGroup/Subgroup1/Subgroup2/repo

```
degit server.dot.com/ParentGroup/Subgroup1/Subgroup2/repo test
! Command failed: git clone --depth 1 git@server.dot.com:ParentGroup/Subgroup1 test/.degitTyBtMP
Cloning into 'test/.degitTyBtMP'...
remote:
remote: ========================================================================
remote:
remote: The project you were looking for could not be found or you don't have permission to view it.
remote:
remote: ========================================================================
remote:
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

So it only "reads" the first subgroup and thinks it is indeed the repo, is possible to achieve what i'm trying to do?

Cheers!

**Comment by nake89** [comment]
Hi! 🙂 

I think maybe I could add an option for this. E.g.
`degit --subgroup server.dot.com/ParentGroup/Subgroup1/Subgroup2/repo test`
and/or
`degit -s server.dot.com/ParentGroup/Subgroup1/Subgroup2/repo test`

Would this be ok?

I would rather not modify the current parsing to make it work automatically (unless somebody gives me a PR and gives me a subgroup repo I can test this with). (Maybe I'll change my mind after looking at the code :D )

Is it possible to do these subgroups with GitHub so I could test this functionality?

**Comment by dark452** [comment]
Hello @nake89  many thanks for your answer. 

Sure, that would completely amazing, either of both options are perfect and solves my problem.

I can provide this groups and subgroups with GitLab, i will take a look to GitHub, in order to check if is possible to do something similar.

**Comment by dark452** [comment]
@nake89 I've created this repo that simulates the same directory structure (Gitlab):

[Link](https://gitlab.com/redeshostcl/azure/infrastructure/test-repo)


Thanks for provide the possibility to help me out with this issue.

**Comment by nake89** [comment]
I got it to work with https, but with mode=git not. TBH I'm fine with that.
My PR: https://github.com/tiged/tiged/pull/24
I still have to add a test and also manually test it. But seems to be working:
```
➜  tiged git:(23-gitlab-subgroups) node bin.js --subgroup gitlab.com/redeshostcl/azure/infrastructure/test-repo subgroup-test
> cloned redeshostcl/azure#HEAD to subgroup-test
➜  tiged git:(23-gitlab-subgroups) ✗ ls -la subgroup-test
total 8
drwxrwxr-x   3 kivi.local  staff   96 Jul 14 17:09 .
drwxr-xr-x  24 kivi.local  staff  768 Jul 14 17:09 ..
-rw-r--r--   1 kivi.local  staff  703 Jul  7 21:16 main.tf
➜  tiged git:(23-gitlab-subgroups) ✗ head subgroup-test/main.tf
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "2.46.1"
    }
  }
}

provider "azurerm" {
➜  tiged git:(23-gitlab-subgroups) ✗
```

**Comment by dark452** [comment]
@nake89 great!! thanks for your help with this new feature :)

**Comment by nake89** [comment]
Feature now published on version 2.11.0
https://www.npmjs.com/package/tiged/v/2.11.0

**Comment by dark452** [comment]
@nake89 I already updated tiged and i can confirm the new feature is working like a charm.
I've even tested it with the following structure:

```
degit -s https://gitlab.com/Group/SubGroup1/SubGroup2/SubGroup3/SubGroup4/repo 4sg-structure
```

Thank you!
